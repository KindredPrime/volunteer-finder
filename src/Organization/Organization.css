.Organization a {
  color: var(--accent-color);
}

.Organization button {
  margin-top: 10px;
}

.Organization__contact-info p {
  text-align: center;
}

.Organization__causes h2 {
  margin-bottom: 10px;
}

.Organization__causes-list {
  margin-top: 0px;
}

.Organization__causes-list > li {
  margin-bottom: 5px;
}