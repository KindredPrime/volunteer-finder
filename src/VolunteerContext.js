import React from 'react';

const VolunteerContext = React.createContext({
  orgs: [],
  causes: [],
  addOrg: () => {},
  deleteOrg: () => {},
  appError: null,
  // Is the app fetching data from the API?
  fetching: false
});

export default VolunteerContext;