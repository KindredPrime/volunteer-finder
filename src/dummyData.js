const dummyUsers = [
  {
    id: 1,
    email: 'email1@email.com',
    username: 'user1',
    password: 'password1',
    orgsAdded: [1],
    eventsAdded: [1, 2]
  },
  {
    id: 2,
    email: 'email2@email.com',
    username: 'user2',
    password: 'password2',
    orgsAdded: [2],
    eventsAdded: [3]
  },
  {
    id: 3,
    email: 'email3@email.com',
    username: 'user3',
    password: 'password3',
    orgsAdded: [3],
    eventsAdded: []
  },
  {
    id: 4,
    email: 'email4@email.com',
    username: 'user4',
    password: 'password4',
    orgsAdded: [],
    eventsAdded: []
  },
  {
    id: 5,
    email: 'email5@email.com',
    username: 'user5',
    password: 'password5',
    orgsAdded: [4],
    eventsAdded: [4]
  }
];

const dummyOrgs = [
  {
    id: 1,
    name: 'YMCA',
    website: 'https://www.ymca.com',
    phone: '1-800-555-7777',
    email: 'contact@ymca.com',
    address: '1 YMCA Street Real City, Real State',
    causes: [1, 2, 3],
    tags: [2, 3, 4, 6]
  },
  {
    id: 2,
    name: 'ASPCA',
    website: 'https://www.aspca.com',
    phone: '1-800-123-4567',
    email: 'contact@aspca.com',
    address: '1 ASPCA Street Real City, Real State',
    causes: [5, 2],
    tags: [2, 3, 4, 10]
  },
  {
    id: 3,
    name: 'National VOAD',
    website: 'https://www.nvoad.com',
    phone: '1-800-456-7890',
    email: 'contact@nvoad.com',
    address: '1 NVOAD Street Rivendell, Middle Earth',
    causes: [4, 2],
    tags: [2, 3, 4, 5, 7, 11]
  },
  {
    id: 4,
    name: 'National Air and Space Museum',
    website: 'https://www.airandspace.com',
    phone: '1-800-987-6543',
    email: 'contact@airandspace.com',
    address: '1 Space Street Air City, Outer Space',
    causes: [6, 7],
    tags: [1, 2, 3, 4, 6, 12]
  }
];

const dummyEvents = [
  {
    id: 1,
    name: 'Member Appreciation Day',
    organization: 1,
    location: 'Reston YMCA Center',
    date: 'February 20, 2020 00:00:00',
    duration: '4 hours',
    description: 'Greet YMCA members as they enter the facility and offer them snacks.  See https://www.website.com for more details.',
    causes: [2, 3],
    tags: [2, 4, 12]
  },
  {
    id: 2,
    name: 'Triathlon Assistance',
    organization: 1,
    location: 'Reston YMCA Center',
    date: 'January 27, 2020 00:00:00',
    duration: '4 hours',
    description: 'Help YMCA staff host their annual indoor triathlon: sign in triathlon participants, direct the participants to the different stages of the triathlon, and record times. See https://www.triathlon.com for more details.',
    causes: [2, 3],
    tags: [2, 3, 9]
  },
  {
    id: 3,
    name: 'Adoption Day',
    organization: 2,
    location: 'Times Square, New York City, New York',
    date: 'December 13, 2020 00:00:00',
    duration: '8 hours',
    description: 'Help the ASPCA run their annual adoption day for their local pet shelter.',
    causes: [5],
    tags: [2, 3, 9, 12]
  },
  {
    id: 4,
    name: 'Docent Training Class',
    organization: 4,
    location: '14390 Air and Space Museum Pkwy Chantilly, VA 20151',
    date: 'January 30, 2021 00:00:00',
    duration: '2 hours',
    description: 'This is a required class for anyone who wants to be a docent (tour guide) at the National Air and Space Museum.',
    causes: [6, 7],
    tags: [2, 3, 13]
  }
];

const dummyTags = [
  {
    id: 1,
    name: 'virtual'
  },
  {
    id: 2,
    name: 'in-person'
  },
  {
    id: 3,
    name: 'weekends'
  },
  {
    id: 4,
    name: 'weekdays'
  },
  {
    id: 5,
    name: 'nights'
  },
  {
    id: 6,
    name: 'group'
  },
  {
    id: 7,
    name: 'delivery'
  },
  {
    id: 8,
    name: 'cooking'
  },
  {
    id: 9,
    name: 'administration'
  },
  {
    id: 10,
    name: 'cleaning'
  },
  {
    id: 11,
    name: 'heavy lifting'
  },
  {
    id: 12,
    name: 'greeting'
  },
  {
    id: 13,
    name: 'training'
  }
];

const dummyCauses = [
  {
    id: 1,
    name: 'Youth'
  },
  {
    id: 2,
    name: 'Community'
  },
  {
    id: 3,
    name: 'Health'
  },
  {
    id: 4,
    name: 'Disaster Response'
  },
  {
    id: 5,
    name: 'Animals'
  },
  {
    id: 6,
    name: 'Museums'
  },
  {
    id: 7,
    name: 'Education'
  },
  {
    id: 8,
    name: 'Human Rights'
  },
  {
    id: 9,
    name: 'Arts'
  }
];

export {
  dummyUsers,
  dummyOrgs,
  dummyEvents,
  dummyTags,
  dummyCauses
};