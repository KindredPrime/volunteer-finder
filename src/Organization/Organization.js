import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import VolunteerContext from '../VolunteerContext';
import Nav from '../Nav/Nav';
import EventList from '../EventList/EventList';

function Organization(props) {
  const orgId = props.match.params.id;

  return (
    <VolunteerContext.Consumer>
      {(value) => {
        const { users, orgs, events } = value;
        const org = orgs.find((elem) => elem.id === parseInt(orgId));

        if (org) {
          const { name, website, phone, email, address, description, causes, tags } = org;

          const orgEvents = events.filter((event) => event.organization === name);

          const creator = users.find((user) => user.orgsAdded.includes(name));

          return (
            <div className="Organization">
            <Nav />
    
            <header>
              <h1>{name}</h1>
            </header>

            <section>
              <header>
                <h2>Contact Info</h2>
              </header>

              <p>Website: <a href={website} target="_blank" rel="noreferrer">{website}</a></p>
              <p>Phone: {phone}</p>
              <p>Email: {email}</p>
              <p>Address: {address}</p>
              <p>{description}</p>
            </section>

            <section>
              <header>
                <h2>Causes</h2>
              </header>
              
              <ul className="Organization__causes">
                {causes.map((cause, index) => (
                  <li key={index}>
                    {cause}
                  </li>
                ))}
              </ul>
            </section>

            <section>
              <header>
                <h2>Tags</h2>
              </header>

              <p className="Organization__tags">{tags.join(', ')}</p>
            </section>

            <section>
              <header>
                <h2>Events</h2>
              </header>

              <EventList events={orgEvents} />
            </section>
            
            <p>Created By: <Link to={`/user/${creator.id}`}>{creator.username}</Link></p>
          </div>
          );
        }

        return (
          <div className="Organization">
            <Nav />

            <p>No organization found</p>
          </div>
        )
      }}
    </VolunteerContext.Consumer>
  );
}

Organization.propTypes = {
  match: PropTypes.shape({
    params: PropTypes.shape({
      id: PropTypes.string
    })
  }).isRequired
};

export default Organization;