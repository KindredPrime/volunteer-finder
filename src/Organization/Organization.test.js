import ReactDOM from 'react-dom';
import { BrowserRouter } from 'react-router-dom';
import { render } from '@testing-library/react';
import Organization from './Organization';
import { dummyOrgs, dummyCauses } from '../dummyData';
import VolunteerContext from '../VolunteerContext';

describe('Organization Component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <BrowserRouter>
        <Organization match={{
          params: {
            id: '0'
          }
        }} />
      </BrowserRouter>,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI as expected', () => {
    const contextValue = {
      orgs: dummyOrgs,
      causes: dummyCauses
    };
    render(
      <VolunteerContext.Provider value={contextValue}>
        <BrowserRouter>
          <Organization match={{
            params: {
              id: '1'
            }
          }} />
        </BrowserRouter>
      </VolunteerContext.Provider>
    );

    expect(document.body).toMatchSnapshot();
  });
});