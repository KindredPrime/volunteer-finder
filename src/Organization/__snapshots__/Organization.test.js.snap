// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Organization Component renders the UI as expected 1`] = `
<body>
  <div>
    <main
      class="Organization"
    >
      <header>
        <h1>
          YMCA DC
        </h1>
      </header>
      <section
        class="Organization__contact-info"
      >
        <header>
          <h2>
            Contact Info
          </h2>
        </header>
        <p>
          Website: 
          <a
            href="https://www.ymcadc.org/locations/ymca-fairfax-county-reston/"
            rel="noreferrer"
            target="_blank"
          >
            https://www.ymcadc.org/locations/ymca-fairfax-county-reston/
          </a>
        </p>
        <p>
          Phone: 
          703-742-8800
        </p>
        <p>
          Email: 
          Carson.Henry@ymcadc.org
        </p>
        <p>
          Address: 
          12196 Sunset Hills Road, Reston, VA 20190
        </p>
      </section>
      <section>
        <header>
          <h2>
            About 
            YMCA DC
          </h2>
        </header>
        <p
          class="Organization__description"
        >
          Today, the Y engages more than 10,000 neighborhoods across the U.S. As the nation’s leading nonprofit committed to helping people and communities to learn, grow and thrive, our contributions are both far-reaching and intimate—from influencing our nation’s culture during times of profound social change to the individual support we provide an adult learning to read.

    By empowering young people to reach their full potential, improving individual and community well-being and giving back and inspiring action in our communities, the Y ensures that everyone has the opportunity to become healthier, more confident, connected and secure.
        </p>
      </section>
      <section
        class="Organization__causes"
      >
        <header>
          <h2>
            Causes
          </h2>
        </header>
        <ul
          class="Organization__causes-list"
        >
          <li>
            Youth
          </li>
          <li>
            Health
          </li>
          <li>
            Elderly
          </li>
        </ul>
      </section>
      <button
        type="button"
      >
        Delete Organization
      </button>
    </main>
  </div>
</body>
`;
